src/LiveQuery/ParseLiveQueryServer.js

```diff
@@ -581,6 +579,24 @@ class ParseLiveQueryServer {
       });
   }
 
+  async getAuthFromClient(client: any, res: any, requestId: number, sessionToken: string) {
+    const getSessionFromClient = () => {
+      const subscriptionInfo = client.getSubscriptionInfo(requestId);
+      if (typeof subscriptionInfo === 'undefined') {
+        return client.sessionToken;
+      }
+      return subscriptionInfo.sessionToken || client.sessionToken;
+    };
+    if (!sessionToken) {
+      sessionToken = getSessionFromClient();
+    }
+    const { auth } = await this.getAuthForSessionToken(sessionToken);
+    if (auth && auth.user) {
+      res.user = auth.user;
```

Comment:
Is this method actually modifying the passed `req` object? If so, I think that is pretty difficult to infer, because the method looks like its purpose is just to return an object.

Reply:
Yes it does. How would you propose to write it? should I return `auth` and then set `res.user`?

Reply:
Yes, I think that would be clearer. A "hidden" modification in a method named `get...` may be hard to debug in the future.